src/components/elements/MapSidePanel/MapSidePanel.tsx

```diff
@@ -55,9 +55,35 @@ const MapSidePanel = ({
   const [openMenu, setOpenMenu] = useState(false);
   const [clickedButton, setClickedButton] = useState<string>("");
   const checkboxRefs = useRef<HTMLInputElement[]>([]);
+
+  const [currentPage, setCurrentPage] = useState(1);
+  const pageSize = 20;
+
   const { isMonitoring, setEditPolygon, setIsUserDrawingEnabled } = useMapAreaContext();
   const { map } = mapFunctions;
 
+  const filteredItems = useMemo(() => {
+    if (checkedValues.length === 0) {
+      return items;
+    }
+    return items.filter(item => checkedValues.includes(item.status));
+  }, [items, checkedValues]);
+
+  const totalPages = Math.max(1, Math.ceil(filteredItems.length / pageSize));
+  const startIndex = (currentPage - 1) * pageSize;
+  const endIndex = startIndex + pageSize;
+  const currentPageItems = filteredItems.slice(startIndex, endIndex);
+
+  useEffect(() => {
+    if (currentPage > totalPages && totalPages > 0) {
+      setCurrentPage(totalPages);
+    }
+  }, [currentPage, totalPages]);
+
+  useEffect(() => {
+    setCurrentPage(1);
+  }, [checkedValues]);
```

Comment:
I'm seeing a lot of duplicated code in this component from the one above. Please create a hook to manage this pagination approach and use in both places.